node_modules/oae-principals/tests/test-guests.js

```diff
@@ -0,0 +1,437 @@
+                    assert.ok(!err);
+
+                    // Make a guest a viewer of the content item through the content permissions API
+                    var email = TestsUtil.generateTestEmailAddress();
+                    var roleChanges = {};
+                    roleChanges[email] = 'member';
+                    RestAPI.Group.setGroupMembers(simong.restContext, group.id, roleChanges, function(err) {
+                        assert.ok(!err);
+                    });
+                    
+                    // An email should've been sent to the guest
+                    EmailAPI.once('debugSent', function(message) {
+                        // Verify we sent an email to the invited user
+                        assert.strictEqual(message.to[0].address, email);
+
+                        // Verify a token is passed in the email
+                        var token = message.text.match(/\?token=([a-zA-Z0-9]{32})/)[1];
+                        assert.ok(token);
+
+                        // Sign in as the guest
+                        PrincipalsTestUtil.signIntoGuestAccountWithToken(token, function(loggedinGuest) {
+
+                            // The guest should have access to the piece of content
+                            RestAPI.Group.getGroup(loggedinGuest.restContext, group.id, function(err, group) {
+                                assert.ok(!err);
+
+                                // Verify the guest is in the members list
+                                RestAPI.Group.getGroupMembers(simong.restContext, group.id, null, null, function(err, members) {
+                                    assert.ok(!err);
+                                    assert.strictEqual(members.results.length, 2);
+                                    var guest = _.findWhere(members.results, {'role': 'member'});
+                                    assert.ok(guest);
+
+                                    // Make the guest a manager
+                                    var roleChanges = {};
+                                    roleChanges[email] = 'manager';
+                                    RestAPI.Group.setGroupMembers(simong.restContext, group.id, roleChanges, function(err) {
+                                        assert.ok(!err);
+
+                                        // Verify the guest is a manager
+                                        RestAPI.Group.getGroup(loggedinGuest.restContext, group.id, function(err, group) {
+                                            assert.ok(!err);
+                                            assert.ok(group.isManager);
+                                            return callback();
+                                        });
+                                    });
+                                });
+                            });
+                        });
+                    });
+                });
+            });
+        });
+    });
```

Comment:
Wouldn't we always have to do `2` when a new user signs in to prevent 2 people from having the same email address?
What do we do when a new user signs in and a user with that email address already exists on a different tenant? Do we have tests for these scenarios?